src/NLog.Targets.Syslog/MessageSend/Tcp.cs

```diff
@@ -66,7 +67,7 @@ private bool IsSocketConnected()
             if (connectionCheckTimeout <= 0)
                 return true;
 
-            return  tcp.Client.Poll(connectionCheckTimeout, SelectMode.SelectRead) && tcp.Client.Available == 0;
+            return  tcp.Client.Poll(connectionCheckTimeout, SelectMode.SelectWrite) && tcp.Client.Available == 0;
```

Comment:
Thanks for looking at my PR, the blog in particular is very helpful!

I made this change to because the socket was being torn down and setup for every message as `IsSocketConnected` was always returning false. I decided to change it on this PR because on my rsyslog server it was causing the following and felt it was important to change:
```
Mar 12 21:33:03 DESKTOP-3JEJ4K2 TestApp.FormTest[1784] ﻿This is a sample info message
Mar 12 21:33:16 muzzell-rsyslog rsyslogd-2078: unexpected GnuTLS error -110 in nsd_gtls.c:530: The TLS connection was non-properly terminated.  [v8.16.0 try http://www.rsyslog.com/e/2078 ]
Mar 12 21:33:16 muzzell-rsyslog rsyslogd-2078: netstream session 0x7fdd780121d0 from 10.8.0.22 will be closed due to error  [v8.16.0 try http://www.rsyslog.com/e/2078 ]
Mar 12 21:33:05 DESKTOP-3JEJ4K2 TestApp.FormTest[1784] ﻿This is a sample info message
Mar 12 21:33:18 muzzell-rsyslog rsyslogd-2078: unexpected GnuTLS error -110 in nsd_gtls.c:530: The TLS connection was non-properly terminated.  [v8.16.0 try http://www.rsyslog.com/e/2078 ]
Mar 12 21:33:18 muzzell-rsyslog rsyslogd-2078: netstream session 0x7fdd78011c40 from 10.8.0.22 will be closed due to error  [v8.16.0 try http://www.rsyslog.com/e/2078 ]
Mar 12 21:33:08 DESKTOP-3JEJ4K2 TestApp.FormTest[1784] ﻿This is a sample info message
Mar 12 21:33:21 muzzell-rsyslog rsyslogd-2078: unexpected GnuTLS error -110 in nsd_gtls.c:530: The TLS connection was non-properly terminated.  [v8.16.0 try http://www.rsyslog.com/e/2078 ]
Mar 12 21:33:21 muzzell-rsyslog rsyslogd-2078: netstream session 0x7fdd78011de0 from 10.8.0.22 will be closed due to error  [v8.16.0 try http://www.rsyslog.com/e/2078 ]
Mar 12 21:33:10 DESKTOP-3JEJ4K2 TestApp.FormTest[1784] ﻿This is a sample info message
```

Is that tear-down intentional, or do you think is my SSL setup getting in the way somehow?